direct/core/src/main/java/cz/o2/proxima/direct/batch/BatchLogReaders.java

```diff
@@ -120,9 +132,21 @@ public ThroughputLimitedBatchLogObserver(
       this.limiter = Objects.requireNonNull(limiter);
     }
 
+    @Override
+    public void onCompleted() {
+      limiter.close();
```

Comment:
is throwing error here ok?

Reply:
I suppose it is. But we might swap the order of commands, so that the observer is notified and only after that the limiter is closed. Will do that.